### PR TITLE
WIP Create New CI Environment

### LINKED ISSUE #218: In Memory Testing
## Background

Copied from Matt's comment below:

Just in case it helps, the functional requirement is to be about to set up a working hackerspace instance (at least, insofar is required to run unit regress tests) from nothing but the source code from git.

Technically, this means being dropped onto a clean Ubuntu environment with some basic tools preinstalled, cloning the GitHub repo, doing a bundle install, and then saying "run unit tests and exit non-zero on failure".
I think the best option for this last bit is having a rails environment that sets up an in-memory sqlite3 db, such that this environment's settings won't interfere with any other deployments, but am open to better ideas.

### PR DESCRIPTION
**Pull request by cassar**
Fixes #218 

'continuous-integration' was too hard to keep typing so it's now 'ci' 😅

Next step is to get `$ rails test -e ci` working.

### Pull Request Checklist

* [ ] Pull request is based on the `main` branch
* [ ] Pull request includes a [sign off](../CONTRIBUTING.md#sign-off)

### CODE REVIEW

File: config/database.yml [hunk 1]
```diff
@@ -62,6 +62,15 @@ test:
   password: <%= ENV['POSTGRES_PASSWORD'] %>
   host: <%= ENV['POSTGRES_HOST'] || 'localhost' %>
 
+# Warning: The database defined as "ci" will be erased and
+# re-generated from your development database when you run "rake".
+# Do not set this db to the same as development or production.
+ci:
+  database: db/ci.sqlite3
```
Comment (mattcen) [inline config/database.yml:69]: Did you try `database: ":memory:"`[^1] here?

[^1]: https://stackoverflow.com/questions/54611220/launch-rspec-tests-with-sqlite-memory-migrations-are-pending

**Review by mattcen — APPROVED** [review]
All the code you've written makes sense, but given I'm still unclear quite how to specifically use different Rails environments, I'm not sure how to test it.

File: config/database.yml (hunk 1 above)
Comment (mattcen) [reply config/database.yml:69]: Also: https://philippe.bourgau.net/5-minutes-hack-to-speed-up-rspec-in-rails-5-using-in-memory-sqlite/

Comment (cassar) [reply config/database.yml:69]: ooo thanks, will give that a go next. 👍

Comment (cassar) [reply config/database.yml:69]: And this [In memory sqlite database for rails testing](http://chrisroos.co.uk/blog/2006-02-08-in-memory-sqlite-database-for-rails-testing)

**Comment by cassar** [comment]
@mattcen , I'm going to have to jump off this, have played around using the articles above and had no luck.

Based on the article it appears that a separate environment is actually not the way to go.

Happy to troubleshoot, but I'm not terrible familiar with much outside of the default config, I'm usually just happy once the default start working 😅

**Pull request closed without merging**